docs/conf.py

```diff
@@ -0,0 +1,247 @@
+# Configuration file for the Sphinx documentation builder.
+#
+# This file only contains a selection of the most common options. For a full
+# list see the documentation:
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+# -- Path setup --------------------------------------------------------------
+
+# If extensions (or modules to document with autodoc) are in another directory,
+# add these directories to sys.path here. If the directory is relative to the
+# documentation root, use os.path.abspath to make it absolute, like shown here.
+#
+# import os
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = u'SymbiFlow examples'
+authors = u'SymbiFlow'
+copyright = authors + u', 2020'
+
+# -- General configuration ---------------------------------------------------
+
+# Add any Sphinx extension module names here, as strings. They can be
+# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
+# ones.
+extensions = [
+    'sphinx_tabs.tabs',
+    'sphinxcontrib.jinja',
+]
+
+# Add any paths that contain templates here, relative to this directory.
+templates_path = ['_templates']
+
+# List of patterns, relative to source directory, that match files and
+# directories to ignore when looking for source files.
+# This pattern also affects html_static_path and html_extra_path.
+exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store']
+
+
+# -- Options for HTML output -------------------------------------------------
+
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+#
+
+html_show_sourcelink = True
+html_sidebars = {
+    "**": ["logo-text.html", "globaltoc.html", "localtoc.html", "searchbox.html"]
+}
+
+html_theme = 'sphinx_material'
+html_theme_options = {
+    'nav_title': project,
+    'color_primary': 'deep-purple',
+    'color_accent': 'purple',
+    'repo_name': "symbiflow/symbiflow-examples",
+    'repo_url': 'https://github.com/symbiflow/symbiflow-examples',
+    'globaltoc_depth': 2,
+    'globaltoc_collapse': True
+}
+
+# Add any paths that contain custom static files (such as style sheets) here,
+# relative to this directory. They are copied after the builtin static files,
+# so a file named "default.css" will overwrite the builtin "default.css".
+# html_static_path = ['_static']
+
+import os
+from docutils.core import publish_doctree
+
+
+full_name_lut = {
+    'a35t': 'Arty 35T',
+    'a100t': 'Arty 100T',
+    'basys3': 'Basys 3',
+    'eos_s3': 'EOS S3',
+}
+families = ('xc7', 'eos-s3')
+
+
+def handle_default_with_literals(block):
```

Comment:
Would be good to have some short docstrings for all these functions.

Reply:
added